li-hadoop-plugin/src/main/groovy/com/linkedin/gradle/liazkaban/LiAzkabanDslCompiler.groovy

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.gradle.liazkaban;
+
+
+import com.linkedin.gradle.azkaban.AzkabanDslCompiler;
+import com.linkedin.gradle.hadoopdsl.job.Job;
+import com.linkedin.gradle.libangbang.LiHadoopShellCommand;
+import com.linkedin.gradle.libangbang.LiHadoopShellCommandFactory;
+import com.linkedin.gradle.lihadoopdsl.lijob.LiBangBangJob;
+import com.linkedin.gradle.lihadoopdsl.lijob.LiPigBangBangJob;
+import groovy.text.GStringTemplateEngine;
+import groovy.text.Template;
+import org.gradle.api.Project;
+
+class LiAzkabanDslCompiler extends AzkabanDslCompiler {
+
+  private static final String BANGBANG_TEMPLATE = "LiBangBangJob.template";
+  private LiHadoopShellCommandFactory bangBangCommandFactory;
+
+  /**
+   * Constructor for the AzkabanDslCompiler.
+   *
+   * @param project The Gradle project
+   */
+  LiAzkabanDslCompiler(Project project) {
+    super(project);
```

Comment:
Seems like you should set bangBangCommandFactory = new LiHadoopShellCommandFactory() once in the constructor after the super(project) call instead of setting it every time later